src/main/kotlin/com/terraformation/backend/tracking/ObservationService.kt

```diff
@@ -408,38 +412,56 @@ class ObservationService(
   }
 
   /**
-   * Schedule an ad-hoc observation. This creates an ad-hoc observation, creates an ad-hoc
-   * monitoring plot, and adds the plot to the observation.
+   * Record an ad-hoc observation. This creates an ad-hoc observation, creates an ad-hoc monitoring
+   * plot, adds the plot to the observation, and complete the observation with plants.
    */
-  fun scheduleAdHocObservation(
-      endDate: LocalDate,
+  fun completeAdHocObservation(
+      conditions: Set<ObservableCondition>,
+      notes: String?,
+      observedTime: Instant,
       observationType: ObservationType,
       plantingSiteId: PlantingSiteId,
+      plants: Collection<RecordedPlantsRow>,
       plotName: String,
-      startDate: LocalDate,
       swCorner: Point,
   ): Pair<ObservationId, MonitoringPlotId> {
     requirePermissions { scheduleAdHocObservation(plantingSiteId) }
 
-    validateSchedule(plantingSiteId, startDate, endDate)
+    if (observedTime.isAfter(clock.instant())) {
```

Comment:
It's good to reject wildly wrong timestamps, but I'm concerned this check will be sensitive to clock skew and will reject legitimate observations where the user's phone clock is ahead of the server's clock by a small amount.

Reply:
I will add it a buffer margin!